scvi/external/gimvi/_model.py

```diff
@@ -519,6 +520,61 @@ def load(
         model.to_device(device)
         return model
 
+    @staticmethod
+    def setup_anndata(
+        adata: AnnData,
+        batch_key: Optional[str] = None,
+        labels_key: Optional[str] = None,
+        categorical_covariate_keys: Optional[List[str]] = None,
+        continuous_covariate_keys: Optional[List[str]] = None,
+        copy: bool = False,
+    ) -> Optional[AnnData]:
+        """
+        Sets up the :class:`~anndata.AnnData` object for this model.
+
+        A mapping will be created between data fields used by this model to their respective locations in adata.
+        This method will also compute the log mean and log variance per batch for the library size prior.
+
+        None of the data in adata are modified. Only adds fields to adata.
+
+        Parameters
+        ----------
+        adata
```

Comment:
Resuable documentation would be really useful for these. Relevant notion task: https://www.notion.so/See-if-towncrier-can-automate-release-notes-92866f9e1e6b48218b4690b3041370b7

Reply:
agree but the documentation is not exactly the same here contrary to what is the case with doc_differential_expression @adamgayoso

Reply:
added in docrep per Adam's suggestion